### PR TITLE
Noop usage of `esm` module

### LINKED ISSUE #2569: Deploy failure with ESM
**Impact**: This completely breaks `now.sh` for me. I won't be able to deploy new builds until we get this fixed.

**Update:** I am no longer depending on the module mentioned below, but am still getting errors about esm. Apparently, something else is also depending on esm.

~~I just created [a new npm module](https://github.com/ericelliott/gql-validate) that uses esm~~, and then tried to deploy a project that uses the module. However, the deployment is failing with the following error output:

> ./node_modules/gql-validate/node_modules/esm/esm.js Critical dependency: require function is used in a way in which dependencies cannot be statically extracted  ./node_modules/gql-validate/node_modules/esm/esm.js Module not found: Can't resolve 'internal/bootstrap/loaders' in '/tmp/634390ac/node_modules/gql-validate/node_modules/esm'
--
Jul 17 2019 11:43:47:701 | ...package.json | > Build error occurred
Jul 17 2019 11:43:47:703 | ...package.json | { TypeError: Cannot destructure property `dir` of 'undefined' or 'null'.     at Object.<anonymous> (/tmp/634390ac/.next/serverless/pages/register.js:9022:1704)     at Object.eqrp (/tmp/634390ac/.next/serverless/pages/register.js:9023:30)     at __webpack_require__ (/tmp/634390ac/.next/serverless/pages/register.js:23:31)     at Object.<anonymous> (/tmp/634390ac/.next/serverless/pages/register.js:9011:11)     at Object.epOS (/tmp/634390ac/.next/serverless/pages/register.js:9015:30)     at __webpack_require__ (/tmp/634390ac/.next/serverless/pages/register.js:23:31)     at Module.edYE (/tmp/634390ac/.next/serverless/pages/register.js:8825:20)     at __webpack_require__ (/tmp/634390ac/.next/serverless/pages/register.js:23:31)     at /tmp/634390ac/.next/serverless/pages/register.js:91:18     at Object.<anonymous> (/tmp/634390ac/.next/serverless/pages/register.js:94:10) type: 'TypeError', '$error': '$error' }

See [this issue in standard-things/esm](https://github.com/standard-things/esm/issues/831).

### PR DESCRIPTION
**Pull request by Timer**
tl;dr: The [`esm`](https://github.com/standard-things/esm) package works great for the Node world, however, is totally unnecessary in a webpack world.
It has the same behavior of webpack but breaks static analysis.

---

Webpack code that's bundled for consumption in Node can never favor the `module` key over the `main` key, as the import semantics of CommonJS and ES Modules are not interoperable / magically compatible.

More background here: https://github.com/webpack/webpack/issues/4742

Meaning, the following code:
```js
const M = require('my-module');
```

is not equivalent to:
```js
import M from 'my-module'
```

but instead, is equivalent to:
```js
import * as M from 'my-module'
```

This means code that's instructed to unwittingly bundle against an ESM version of package suddenly becomes incompatible, as their export has moved to the `default` key.

You could argue that it's a package's responsibility to ensure that their CommonJS version exports using the `default` key:
```js
module.exports = function() { } // "WRONG"

module.exports = {
  default: function() { }
} // "CORRECT"
```

> Note: ☝️ this is the behavior of [`esm`](https://github.com/standard-things/esm), properly so.

However, this is not often seen in practice and is not possible to enforce at scale. It only takes one [incorrectly published package](https://github.com/bitinn/node-fetch) to break everyone. 😄 

As a result, we must continue (for Node packages) to import the `main` key before the `module` key, or fix them on a case-by-case basis.

That brings us to the next point, in a [`esm`](https://github.com/standard-things/esm) project, webpack _does not like_ the following code:
```js
require = require('esm')(module)
module.exports = require('./main.js')
```

Re-assigning `require` confuses webpack and causes it to bail out of its bundling job.

The solution? Delete [`esm`](https://github.com/standard-things/esm) because it already works OOTB in webpack:
```js
module.exports = require('./main.js')
```

☝️ This is what this plugin outputs for the above input.

----

~[**There is an alternative to this PR**](https://github.com/zeit/next.js/pull/8082), which is to favor `module` always and white-list every single broken package in the ecosystem to a fixed variant. This may be favorable?~

Closes https://github.com/zeit/next.js/pull/8082
Closes https://github.com/zeit/now-cli/issues/2569

### CODE REVIEW

**Review by Timer — COMMENTED** [review]
This would need to let Babel compile `node_modules`. Alternatively, it could re-use webpack's acorns parser since it's ran against `node_modules` anyway.

**Comment by Timer** [comment]
Closing in favor of the new `experimental-serverless-trace` target.
